src/System.IO.FileSystem.Watcher/src/System/IO/FileSystemWatcher.OSX.cs

```diff
@@ -411,7 +413,7 @@ private bool CheckIfPathIsNested(string eventPath)
                 {
                     // Check if the parent is the root. If so, then we'll continue processing based on the name.
                     // If it isn't, then this will be set to false and we'll skip the name processing since it's irrelevant.
-                    string parent = System.IO.Path.GetDirectoryName(eventPath);
+                    string parent = System.IO.Path.GetDirectoryName(eventPath) + "/";
```

Comment:
Fixed to use string.Compare instead to avoid allocations. Also fixed elsewhere in the function that compared to the root directly to use this same comparison